expression/integration_test.go

```diff
@@ -8927,6 +8927,14 @@ func (s *testIntegrationSuite) TestClusteredIndexCorCol(c *C) {
 	tk.MustQuery("select (select t2.c_str from t2 where t2.c_str = t1.c_str and t2.c_int = 10 order by t2.c_str limit 1) x from t1;").Check(testkit.Rows("<nil>", "goofy mestorf"))
 }
 
+func (s *testIntegrationSuite) TestJiraSetInnoDBDefaultRowFormat(c *C) {
```

Comment:
just found that we are able to set a noop function even when `tidb_enable_noop_functions` is off:

```sql
TiDB(root@127.0.0.1:test) > show variables like "%noop%";
+----------------------------+-------+
| Variable_name              | Value |
+----------------------------+-------+
| tidb_enable_noop_functions | OFF   |
+----------------------------+-------+
1 row in set (0.03 sec)

TiDB(root@127.0.0.1:test) > set global innodb_default_row_format = "dynamic";
Query OK, 0 rows affected (0.00 sec)
```

could you add some tests to ensure an error is returned when `tidb_enable_noop_functions` is off?

Reply:
Currently variables do not influence `noop` functions. I would like them to be able to, as `noop` sysvars: https://github.com/pingcap/tidb/issues/20357 - we can merge noop sysvars and noop functions if you like, but a lot gets disabled.

In this case however, it's safe to always permit. When JIRA changes the row format to `DYNAMIC`, it has no functional change to TiDB. Changing to `COMPACT` or `REDUNDANT` would have a functional change, but that gets complicated in itself because TiDB defaults to `utf8mb4` and MySQL 5.7 did not. It would have to complain about DDL statements such as an inde on `varchar(255)` due to index length.

Reply:
Another option is that I can just remove `COMPACT` and `REDUNDANT` from the list of settable values. Support for these values is not required.

Reply:
the current implementation looks good to me, we can leave the discussion to https://github.com/pingcap/tidb/issues/20357